tools/check_for_upgrade.sh

```diff
@@ -29,31 +29,36 @@ fi
 # Cancel upgrade if git is unavailable on the system
 whence git >/dev/null || return 0
 
-if [ -f ~/.zsh-update ]
+if mkdir "$ZSH/log/update.lock" 2>/dev/null
```

Comment:
I'd like to have these `if then` statements in the same line, an extra commit converting all these to `if; then` will do.